src/main/java/io/anserini/util/mapper/CountDocumentMapper.java

```diff
@@ -78,53 +46,32 @@ public CountDocumentMapper(MapCollections.Args args) throws Exception {
     }
   }
 
-  public void incrementErrors() {
-    counters.errors.incrementAndGet();
-  }
-
-  public void incrementSkipped() {
-    counters.skipped.incrementAndGet();
-  }
-
-  public void incrementUnindexable() {
-    counters.unindexable.incrementAndGet();
-  }
-
-  public void incrementEmpty() {
-    counters.empty.incrementAndGet();
-  }
-
-  public void incrementIndexed() {
-    counters.indexed.incrementAndGet();
-  }
-
-  public void incrementIndexedBy(int numIndexed) {
-    counters.indexed.addAndGet(numIndexed);
+  @Override
+  public void setContext(DocumentMapperContext context) {
+    this.counters = (CountDocumentMapperContext) context;
   }
 
-  public boolean process(SourceDocument d) {
-    if (!d.indexable()) {
-      incrementUnindexable();
-      return false;
+  @Override
+  public void process(SourceDocument doc, DocumentMapperContext context) {
+    if (!doc.indexable()) {
+      ((CountDocumentMapperContext) context).unindexable.incrementAndGet();
+      return;
     }
 
-    if (whitelistDocids != null && !whitelistDocids.contains(d.id())) {
-      incrementSkipped();
-      return false;
+    if (whitelistDocids != null && !whitelistDocids.contains(doc.id())) {
+      ((CountDocumentMapperContext) context).skipped.incrementAndGet();
+      return;
     }
 
-    incrementIndexed();
-    return true;
+    ((CountDocumentMapperContext) context).processed.incrementAndGet();
```

Comment:
Simplify code by using `counters` instead of `((CountDocumentMapperContext) context)`? And I think the meaning is more clear...